pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -495,65 +499,172 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		interfaceType = vmi.ObjectMeta.Annotations[v1.InterfaceModel]
 	}
 
-	// For now if networks not define in vm specs connect the virtual machine to the pod network
-	if vmi.Spec.Domain.Devices.Interfaces == nil {
-		domain.Spec.Devices.Interfaces = []Interface{{
-			Model: &Model{
-				Type: interfaceType,
-			},
-			Type: "bridge",
-			Source: InterfaceSource{
-				Bridge: DefaultBridgeName,
-			}},
+	networks := map[string]*v1.Network{}
+	for _, network := range vmi.Spec.Networks {
+		networks[network.Name] = network.DeepCopy()
+	}
+
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		net, isExist := networks[iface.Name]
+		if !isExist {
+			return fmt.Errorf("failed to find network %s", iface.Name)
 		}
-	} else {
-		// prepare networks map
-		networks := map[string]*v1.Network{}
-		for _, network := range vmi.Spec.Networks {
-			networks[network.Name] = network.DeepCopy()
+
+		if net.Pod == nil {
+			return fmt.Errorf("network interface type not supported for %s", iface.Name)
 		}
+		// TODO:(ihar) consider abstracting interface type conversion /
+		// detection into drivers
+		if iface.Bridge != nil {
+			domainIface := Interface{
+				Model: &Model{
+					Type: interfaceType,
+				},
+				Type: "bridge",
+				Source: InterfaceSource{
+					Bridge: DefaultBridgeName,
+				},
+				Alias: &Alias{
+					Name: iface.Name,
+				},
+			}
+			domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, domainIface)
+
+		} else if iface.Slirp != nil {
+			// TODO: maybe add interface model to vm spec
+			// Slirp configuration works only with e1000 or rtl8139
```

Comment:
You say only e1000 and rtl work, but then you error on virtio only. There are other options supported by libvirt; should you be explicit and error unless it's e1000 / rtl?

Reply:
I cant find any documentation about all the interfaces that are supported be the slirp interface so I will change the if to check for only e1000 / rtl